react-native/react/actions/login2.js

```diff
@@ -2,51 +2,31 @@
 
 import * as Constants from '../constants/login2'
 import QRCodeGen from 'qrcode-generator'
-import { appendRouteOnUnchanged, navigateTo } from './router'
+import { navigateTo, routeAppend, getCurrentURI, getCurrentTab } from './router'
 import engine from '../engine'
-
-export function login (username, passphrase) {
-  return function (dispatch) {
-    dispatch({
-      type: Constants.login
-    })
-
-    const param = {
-      username,
-      passphrase,
-      storeSecret: true,
-      error: null
-    }
-
-    const incomingMap = {
-      'keybase.1.logUi.log': (param, response) => {
-        console.log(param, response)
-        response.result()
-      }
-    }
-
-    engine.rpc('login.loginWithPassphrase', param, incomingMap, (error, response) => {
-      if (error) {
-        console.log(error)
-      }
-
-      dispatch({
-        type: Constants.loginDone,
-        error: !!error,
-        payload: error || response
-      })
-    })
+import enums from '../keybase_v1'
+import UserPass from '../login2/register/user-pass'
+import PaperKey from '../login2/register/paper-key'
+import CodePage from '../login2/register/code-page'
+import ExistingDevice from '../login2/register/existing-device'
+import SetPublicName from '../login2/register/set-public-name'
+import { switchTab } from './tabbed-router'
+import { DEVICES_TAB } from '../constants/tabs'
+import { loadDevices } from './devices'
+
+export function login () {
+  return (dispatch, getState) => {
+    startLoginFlow(dispatch, getState, enums.provisionUi.ProvisionMethod.device, 'Keybase username', 'lorem ipsum', Constants.loginDone)
   }
 }
 
 export function doneRegistering () {
-  resetCountdown()
   return {
     type: Constants.doneRegistering
   }
 }
 
-export function defaultModeForDeviceRoles (myDeviceRole, otherDeviceRole, brokenMode) {
+function defaultModeForDeviceRoles (myDeviceRole, otherDeviceRole, brokenMode) {
```

Comment:
we were exposing these before but now that we're making these components dynamically the functions aren't exposed